test/preview/components/file_converters/test_azure_ocr_doc_converter.py

```diff
@@ -0,0 +1,100 @@
+import os
+from unittest.mock import patch, Mock
+
+import pytest
+
+from haystack.preview.components.file_converters.azure import AzureOCRDocumentConverter
+
+
+class TestAzureOCRDocumentConverter:
+    @pytest.mark.unit
+    def test_to_dict(self):
+        component = AzureOCRDocumentConverter(endpoint="test_endpoint", credential_key="test_credential_key")
+        data = component.to_dict()
+        assert data == {
+            "type": "AzureOCRDocumentConverter",
+            "init_parameters": {
+                "credential_key": "test_credential_key",
+                "endpoint": "test_endpoint",
+                "id_hash_keys": [],
+                "model_id": "prebuilt-read",
+                "save_json": False,
+            },
+        }
+
+    @pytest.mark.unit
+    def test_from_dict(self):
+        data = {
+            "type": "AzureOCRDocumentConverter",
+            "init_parameters": {
+                "credential_key": "test_credential_key",
+                "endpoint": "test_endpoint",
+                "id_hash_keys": [],
+                "model_id": "prebuilt-read",
+                "save_json": False,
+            },
+        }
+        component = AzureOCRDocumentConverter.from_dict(data)
+        assert component.endpoint == "test_endpoint"
+        assert component.credential_key == "test_credential_key"
+        assert component.id_hash_keys == []
+        assert component.model_id == "prebuilt-read"
+        assert component.save_json is False
+
+    @pytest.mark.unit
+    def test_run(self, preview_samples_path):
+        with patch("haystack.preview.components.file_converters.azure.DocumentAnalysisClient") as mock_azure_client:
+            mock_result = Mock(pages=[Mock(lines=[Mock(content="mocked line 1"), Mock(content="mocked line 2")])])
+            mock_azure_client.return_value.begin_analyze_document.return_value.result.return_value = mock_result
+
+            component = AzureOCRDocumentConverter(endpoint="test_endpoint", credential_key="test_credential_key")
+            output = component.run(paths=[preview_samples_path / "pdf" / "sample_pdf_1.pdf"])
```

Comment:
Out of curiosity, do we need a real file here or a simple string like `test_file.pdf` would work?

Reply:
Ahh right good point